src/libraries/System.Runtime.InteropServices.JavaScript/gen/JSImportGenerator/JSExportGenerator.cs

```diff
@@ -253,8 +254,8 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                             }))
                             .WithBody(Block(
                                 IfStatement(BinaryExpression(SyntaxKind.EqualsExpression,
-                                    IdentifierName("Environment.Version.Major"),
-                                    LiteralExpression(SyntaxKind.NumericLiteralExpression, Literal(7))),
+                                    IdentifierName("System.Runtime.InteropServices.JavaScript.JSHost.GlobalThis"),
```

Comment:
JSHost.GlobalThis is not cheap to evaluate. As I said before, I think we could drop the Net7 support and replace it with something else which will keep the initializer alive until explicitly called by JS.

Reply:
Yes, that is why this is no-merge. I'm not sure what to use. Stopwatch.GetTimestamp is cheaper but ideally we could just protect the target method from trimming with an attribute somehow.